engine/src/main/java/io/camunda/zeebe/engine/processing/job/JobTimeoutTrigger.java

```diff
@@ -14,14 +14,14 @@
 import io.camunda.zeebe.engine.processing.streamprocessor.writers.TypedCommandWriter;
 import io.camunda.zeebe.engine.state.immutable.JobState;
 import io.camunda.zeebe.protocol.record.intent.JobIntent;
-import io.camunda.zeebe.scheduler.ScheduledTimer;
 import java.time.Duration;
 
 public final class JobTimeoutTrigger implements StreamProcessorLifecycleAware {
   public static final Duration TIME_OUT_POLLING_INTERVAL = Duration.ofSeconds(30);
   private final JobState state;
 
-  private ScheduledTimer timer;
+  private boolean timerRunning = false;
```

Comment:
:upside_down_face: I initially thought it meant the timer task was running, so its usage surprised me. But maybe that's just me.

```suggestion
  private boolean timerScheduled = false;
```